undent_example_test.go

```diff
@@ -1,6 +1,7 @@
 package undent_test
```

Comment:
vs _test just as it's not really a test, another option might be put it in cmd/example.go and make it main so it runs.

Reply:
Only reason I made it a separate package name is so the examples contain full external usage code so they render nice on pkg.go.dev.

I did consider changing `undent_test.go` to be `package undent_test` and add examples in there along with the tests. Maybe that's the best approach?